projects/igniteui-angular-elements/src/lib/icon.broadcast.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable, Optional } from '@angular/core';
+import { PlatformUtil } from '../../../igniteui-angular/src/lib/core/utils';
+import { IconMeta, IgxIconService } from '../../../igniteui-angular/src/lib/icon/icon.service';
+
+
+export interface SvgIcon {
+    svg: string;
+    title?: string;
+  }
+
+  export type Collection<T, U> = Map<T, U>;
+
+  export enum ActionType {
+    SyncState = 0,
+    RegisterIcon = 1,
+    UpdateIconReference = 2,
+  }
+
+  export interface BroadcastIconsChangeMessage {
+    actionType: ActionType;
+    collections?: Collection<string, Map<string, SvgIcon>>;
+    references?: Collection<string, Map<string, IconMeta>>;
+  }
+
+/** @hidden @internal **/
+@Injectable()
+export class IgxIconBroadcastService {
+    private iconBroadcastChannel: BroadcastChannel;
+    constructor(
+        protected _iconService: IgxIconService,
+        @Optional() private _platformUtil: PlatformUtil
+    ) {
+        if (this._platformUtil?.isBrowser) {
+            // open broadcast channel for sync with wc icon service.
+            this.iconBroadcastChannel = new BroadcastChannel("ignite-ui-icon-channel");
+            this.iconBroadcastChannel.onmessage = (event) => {
+                const message = event.data as BroadcastIconsChangeMessage;
+                if (message.actionType === ActionType.SyncState ||
+                     message.actionType === ActionType.RegisterIcon) {
+                    this.updateIconsFromCollection(message.collections);
+                }
+
+                if (message.actionType === ActionType.SyncState ||
+                     message.actionType === ActionType.UpdateIconReference) {
+                    this.updateRefsFromCollection(message.references);
```

Comment:
Assuming it's the responsibility of the other side to only emit the icons that do need changing for `UpdateIconReference`?

Reply:
Yes. The other side sends the state of the collections and this side updates its own state.

Reply:
Hehe, then it's not a "yes" answer :) The question was if "the other side to **only** emits the icons that do **need changing**".
If the other side sends the entire state of the collections, **every** API for change on that side will also cause the Angular side to re-register all previous icons for example, which is at least redundant if it doesn't affect performance much.

Reply:
Nvm, checked the implementation, the other side creates a fresh collection for each change, so it is sending just the modified refs. No need to break-check me like that 😄